compiler/luci-interpreter/src/kernels/FullyConnected.test.cpp

```diff
@@ -204,6 +204,56 @@ TEST(FullyConnectedTest, SimpleS4)
               FloatArrayNear(output_data, quantized_tolerance));
 }
 
+TEST(FullyConnectedTest, SimpleU4)
+{
+  std::initializer_list<int32_t> input_shape{1, 2};
+  std::initializer_list<int32_t> weights_shape{4, 2};
+  std::initializer_list<int32_t> bias_shape{4};
+  std::initializer_list<int32_t> output_shape{1, 4};
+  std::initializer_list<float> input_data{
+    1, 3, // batch = 0
+  };
+  std::initializer_list<uint8_t> weights_initializer{
+    8, 9, // unit = 0
+    8, 8, // unit = 1
+    7, 7, // unit = 2
+    8, 8, // unit = 3
+  };
+  std::initializer_list<float> bias_data{0, 1, 2, 3};
+  std::initializer_list<float> output_data{
+    1.5, 1, 0, 3, // batch = 0
+  };
+  std::unique_ptr<IMemoryManager> memory_manager = std::make_unique<TestMemoryManager>();
+
+  Tensor input_tensor =
+    makeInputTensor<DataType::FLOAT32>(input_shape, input_data, memory_manager.get());
+  std::vector<uint8_t> quantized_data(weights_initializer);
+  Tensor weights_tensor(DataType::U4, weights_shape, {{0.5}, {8}}, "");
```

Comment:
Please add another test case for per-channel quantization, i.e., multiple scales and zero points.